src/test/java/org/traccar/protocol/TotemProtocolEncoderTest.java

```diff
@@ -15,10 +15,24 @@ public void testEncode() throws Exception {
 
         Command command = new Command();
         command.setDeviceId(2);
-        command.setType(Command.TYPE_ENGINE_STOP);
+        command.setType(Command.TYPE_REBOOT_DEVICE);
         command.set(Command.KEY_DEVICE_PASSWORD, "000000");
 
-        assertEquals("*000000,025,C,1#", encoder.encodeCommand(command));
+        assertEquals("$$0020CF000000,0061D", encoder.encodeCommand(command));
```

Comment:
Please don't change existing unit tests.

Reply:
I mean I sorta have to, the current unit test with the refactoring won't work. As the TotemProtocolEncoder is used in GPRS communication (and the existing test is wrong for that) and the TotemProtocolSmsEncoder is used for SMS communication  - same style as the other two protocols that I found that have SMS and GPRS Encoders/commands. So that the *ProtocolSmsEncoder class handles SMS, and *ProtocolEncoder handles GPRS.

Reply:
Are you saying that existing encoder was wrong?

Reply:
if I understand how this works, the existing encoder worked for SMS, but for sure not for GPRS.

Reply:
It won't be used for SMS, so sounds like what you are saying is that it was incorrect all along.

Reply:
Why was this marked as resolved? This sounds like a big issue.

Reply:
It was wrong all along. The old test (and command implementation) was using the SMS syntax for GPRS communication. Syntax is different between GPRS and SMS commands (not much, but it is) and I did not even notice that with my changes before (previous PR) - as I did not fully understand how this commands work. 

But current tests and setup is correct for GPRS communication (the TotemProtocolEncoder) and for SMS commands (in TotemProtocolSmsEncoder) with the test example/assertion taken from the documentation.

Reply:
The original implementation is here https://github.com/traccar/traccar/pull/1317. @al3x1s I know it's been a long time, but maybe you can provide some input here. Have you tested commands with SMS or GPRS when you implemented it originally?

Reply:
I will be able to do some tests on real devices (AT05 and AT07) soon-ish, probably early next week. But I'm quite sure that it will work, docs are quite clear and it makes sense that the SMS and GPRS format are different (and that GPRS format for sending commands is close to the format for receiving data, there's also special flag/message for GPRS command response that I need to look into).

Reply:
I have finally found some time to test this, I connected a Totem AT05 and Totem AT07 to the PC, launched Totemtek configuration and log utility and send the command. And it works, log example from the device:
`[2022.02.04, 15:20:36]GPRS REC: $$0020CF000000,0061D

[2022.02.04, 15:20:36]CMD restart
[2022.02.04, 15:20:40]Successful
[2022.02.04, 15:20:40]CMD_CODE_RESTART`
So I can confirm that the original setup/code was wrong and that the current GPRS/SMS command implementation works as tested in this unittest file.